src/rust/engine/fs/src/lib.rs

```diff
@@ -171,13 +240,28 @@ impl PathGlob {
   }
 
   pub fn create(filespecs: &[String]) -> Result<Vec<PathGlob>, String> {
-    let mut path_globs = Vec::new();
+    // Getting a Vec<PathGlob> per filespec is needed to create a `PathGlobs`, but we don't need
+    // that here.
+    let filespecs_globs = Self::spread_filespecs(filespecs)?;
+    let all_globs = Self::flatten_entries(filespecs_globs);
```

Comment:
I'd probably inline `flatten_entries`; it's small enough and only used once that it's more clear to be able to read its body, rather than needing to look around for its implementation